src/foundry/common/abstract/_types.d.mts

```diff
@@ -0,0 +1,246 @@
+export interface DatabaseGetOperation {
+  /**
+   * A query object which identifies the set of Documents retrieved
+   */
+  query: Record<string, any>;
+
+  /**
+   * Get requests are never broadcast
+   */
+  broadcast?: false;
+
+  /**
+   * Return indices only instead of full Document records
+   */
+  index?: boolean;
+
+  /**
+   * An array of field identifiers which should be indexed
+   */
+  indexFields?: string[];
+
+  /**
+   * A compendium collection ID which contains the Documents
+   */
+  pack?: string | null;
+
+  /**
+   * A parent Document within which Documents are embedded
+   */
+  parent?: foundry.abstract.Document.Any;
+
+  /**
+   * A parent Document UUID provided when the parent instance is unavailable
+   */
+  parentUuid?: string;
+}
+
+export interface DatabaseCreateOperation<T extends foundry.abstract.Document.Any = foundry.abstract.Document.Any> {
+  /**
+   * Whether the database operation is broadcast to other connected clients
+   */
+  broadcast: boolean;
+
+  /**
+   * An array of data objects from which to create Documents
+   */
+  data: foundry.data.fields.SchemaField.InnerAssignmentType<T["schema"]["fields"]>[];
+
+  /**
+   * Retain the _id values of provided data instead of generating new ids
+   */
+  keepId?: boolean;
+
+  /**
+   * Retain the _id values of embedded document data instead of generating
+   *    new ids for each embedded document
+   */
+  keepEmbeddedIds?: boolean;
+
+  /**
+   * The timestamp when the operation was performed
+   */
+  modifiedTime?: number;
+
+  /**
+   * Block the dispatch of hooks related to this operation
+   */
+  noHook?: boolean;
+
+  /**
+   * Re-render Applications whose display depends on the created Documents
+   */
+  render?: boolean;
+
+  /**
+   * Render the sheet Application for any created Documents
+   */
+  renderSheet?: boolean;
+
+  /**
+   * A parent Document within which Documents are embedded
+   */
+  parent?: foundry.abstract.Document.Any;
+
+  /**
+   * A compendium collection ID which contains the Documents
+   */
+  pack: string | null;
+
+  /**
+   * A parent Document UUID provided when the parent instance is unavailable
+   */
+  parentUuid?: string | null;
+
+  /**
+   * An alias for 'data' used internally by the server-side backend
+   */
+  _result?: (string | object)[];
```

Comment:
Is `object` the right type here (as it includes arrays and functions) or is `Record<string, unknown>` better?